src/settings/settingsmanager.h

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef settingsmanager_H
+#define settingsmanager_H
+
+#include <QSettings>
+
+#include "loghandler.h"
+#include "setting.h"
+#include "settinggroup.h"
+#include "settingsconnector.h"
+
+/**
+ * @brief The SettingsManager is a singleton class that manages the underlying
+ * QtSetting storage object and serves as the holder of all settings in the
+ * codebase dynamic or static. The lifetime of this singleton is tied to the
+ * lifetime of the QApplication.
+ *
+ * There are two types of settings currently in this codebase:
+ *
+ * * Static settings are settings defined at build time in a declarative manner
+ * e.g. the SETTING macro or the EXPERIMENTAL_FEATURE macro.
+ * * Dynamic settings are settings which are created on demand at runtime.
+ *
+ * The only public constructor for Setting objects is
+ * SettingsManager::instance()->createOrGetSetting, therefore everytime a
```

Comment:
The Settings constructor is public, so it could be created using new Settings(...). However, not a big deal.

Reply:
Yes. I think though, since the only public constructor requires SettingsConnector it might as well be private, because it's just too hard to use outside of the SettingsManager :D